src/core/types/chat.rs

```diff
@@ -36,51 +36,43 @@ impl<'de> serde::de::Visitor<'de> for PrivateChatTypeVisitor {
 fn assert_private_field<'de, D: serde::Deserializer<'de>>(
     des: D,
 ) -> Result<(), D::Error> {
-    des.deserialize_str(PrivateChatTypeVisitor)
+    des.deserialize_str(PrivateChatKindVisitor)
 }
 
-fn serialize_private_field<S: serde::Serializer>(
+/*fn serialize_private_field<S: serde::Serializer>(
     _: &(),
     ser: S,
 ) -> Result<S::Ok, S::Error> {
     ser.serialize_str("private")
-}
+}*/
 
-#[derive(Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
-#[serde(rename_all = "snake_case")]
+#[derive(Debug, Deserialize, Eq, Hash, PartialEq)]
 #[serde(untagged)]
-pub enum ChatType {
-    NotPrivate {
-        #[serde(skip_serializing_if = "Option::is_none")]
+pub enum ChatKind {
+    NonPrivate {
         title: Option<String>,
         #[serde(flatten)]
-        type_: NotPrivateChatType,
-        #[serde(skip_serializing_if = "Option::is_none")]
+        non_private_chat_kind: NonPrivateChatKind,
```

Comment:
This field also should be named just `kind`:
```suggestion
        kind: NonPrivateChatKind,
```